drop-in/src/main/java/com/adyen/checkout/dropin/ui/DropInViewModel.kt

```diff
@@ -162,6 +194,55 @@ class DropInViewModel(private val savedStateHandle: SavedStateHandle) : ViewMode
         )
     }
 
+    fun handleOrderResponse(orderResponse: OrderResponse?) {
+        if (orderResponse == null) {
+            currentOrder = null
+            amount = dropInConfiguration.amount
+            Logger.d(TAG, "handleOrderResponse - Amount reverted: ${amount.value}")
+            Logger.d(TAG, "handleOrderResponse - Order cancelled")
+        } else {
+            currentOrder = orderResponse
+            amount = orderResponse.remainingAmount ?: throw CheckoutException("Provided order does not have a remainingAmount")
+            Logger.d(TAG, "handleOrderResponse - New amount set: ${amount.value}")
+            Logger.d(TAG, "handleOrderResponse - Order cached")
+        }
+    }
+
+    fun updatePaymentComponentStateForPaymentsCall(paymentComponentState: PaymentComponentState<*>) {
+        // include amount value if merchant passed it to the DropIn
+        val amount = when {
+            cachedPartialPaymentAmount != null -> cachedPartialPaymentAmount
+            !amount.isEmpty -> amount
+            else -> null
+        }
+        cachedPartialPaymentAmount = null
+        if (amount != null) {
```

Comment:
since the rest uses `?.let` operator i would prefer using that here instead of null check just for consistency but nbd.